misc/python/materialize/output_consistency/query/query_result.py

```diff
@@ -32,6 +34,14 @@ def __init__(self, query_template: QueryTemplate, query_id: str):
         self.outcomes: List[QueryOutcome] = []
         self.query_template = query_template
 
+    def get_outcome_by_strategy_id(self) -> Dict[str, QueryOutcome]:
+        outcome_by_strategy_id = dict()
+
+        for outcome in self.outcomes:
+            outcome_by_strategy_id[outcome.strategy.identifier] = outcome
+
+        return outcome_by_strategy_id
+
```

Comment:
```suggestion
        return dict([(outcome.strategy.identifier, outcome) for outcome in self.outcomes])
```
(untested) List comprehensions are peak-idiomatic Python code ;)

Reply:
Nice, thanks! 🎉